src/server/controllers/report.controller.js

```diff
@@ -12,12 +13,35 @@ export const renderReportFilters = async (req, res) => {
   } catch (error) {
     logger.error(error);
   }
-  res.render('reports/generateReport', { types: reportTypes, ...req.session });
+  const invalidDateRange = Object.keys(req.query).some(param => param === INVALID_DATE_RANGE);
+  res.render('reports/generateReport', { types: reportTypes, ...req.session, invalidDateRange });
 };
 
+/**
+ * Validate the submitted dates are no later than today and the endDate
+ * is not earlier than the start date.
+ * @param {string} dateFromText End date in the format yyyy-mm-dd
+ * @param {string} dateToText Start date in the format yyyy-mm-dd
+ */
+function validDate(dateFromText, dateToText) {
+  const dateToday = new Date().setHours(0, 0, 0, 0).valueOf();
```

Comment:
We have `moment` as a dependency if you want to use it over native Date API.

Reply:
Yes that would be much better. I'll update it before merging, thanks.